fe/fe-core/src/main/java/com/starrocks/catalog/OlapTable.java

```diff
@@ -907,6 +914,62 @@ public void setColocateGroup(String colocateGroup) {
         this.colocateGroup = colocateGroup;
     }
 
+    public Set<String> getColocateMaterializedViewNames() {
+        return colocateMaterializedViewNames;
+    }
+
+    public void setColocateMaterializedViewNames(Set<String> colocateMaterializedViewNames) {
+        this.colocateMaterializedViewNames = colocateMaterializedViewNames;
+    }
+
+    public boolean isInColocateMvGroup() {
+        return isInColocateMvGroup;
+    }
+
+    public void setInColocateMvGroup(boolean inColocateMvGroup) {
+        this.isInColocateMvGroup = inColocateMvGroup;
+    }
+
+    public void addColocateMaterializedView(String mvName) {
+        colocateMaterializedViewNames.add(mvName);
+    }
+
+    public void removeColocateMaterializedView(String mvName) {
+        if (colocateMaterializedViewNames.contains(mvName)) {
+            if (colocateMaterializedViewNames.size() == 1 && isInColocateMvGroup()) {
```

Comment:
Is this thread-safe when invoke removeColocateMaterializedView？HashSet is not a  thread-safe Collection.

Reply:
the db lock is got in cancelImpl before call the function